src/bun.js/api/server.zig

```diff
@@ -1596,6 +1612,79 @@ pub fn NewServer(protocol_enum: enum { http, https }, development_kind: enum { d
             }
         }
 
+        /// Helper to clear routes for all SNI domains
+        /// This prevents dangling pointers when stopping/reloading servers with SNI
+        fn clearSNIRoutesHelper(this: *ThisServer, app: *App, should_remove_server_names: bool) void {
+            if (comptime !ssl_enabled) return;
+
+            // Preallocate capacity: 1 for main config + count of SNI entries
+            const sni_count = if (this.config.sni) |sni| sni.len else 0;
+            var server_names = std.ArrayList([*:0]const u8).initCapacity(bun.default_allocator, 1 + sni_count) catch std.ArrayList([*:0]const u8).init(bun.default_allocator);
```

Comment:
Please use decl literals instead of writing code like this. Also don't you think we should just emit and error and return if allocation fails instead of almost certainly failing later when we `append`?